### ISSUE TITLE
PorousFlow: replace error with exception in fluid state classes

### ISSUE THREAD

**Issue by cpgr** [idaholab/moose#12018]
## Rationale

The `PorousFlow` fluid state classes are only valid in a limited pressure and/or temperature region. Currently, they error out if pressure and/or temperature is outside this region, which can occur during the solve. It would be nicer if they instead threw an exception, so that the solve could continue with a reduced time step (similar to #10166 in the fluid properties module)

## Description

Basically, replace the `mooseErrors` with `mooseExceptions` in `PorousFlowBrineCO2` and `PorousFlowWaterNCG`

## Impact

Enable simulations to continue if these variables temporarily go out of bounds during the nonlinear solve
